ring-core/test/ring/middleware/test/file.clj

```diff
@@ -93,3 +93,23 @@
       (is (= 200 status))
       (is (= (into #{} (keys headers)) #{"Content-Length" "Last-Modified"}))
       (is (= foo-html body)))))
+
+(deftest test-wrap-file-with-handler-first?-option
+  (let [blog-chp-file (File. ^String public-dir "blog.chp") ;;chp => Cloure Hiccup Page
```

Comment:
Please remove references to `chp`. They make the test seem like it's something to do with your CHP extension idea, when the test is more general.